### PR TITLE
review

### PR DESCRIPTION
**Pull request by arifalkner**

### CODE REVIEW

File: src/classes/BasicAuthAuthorizer.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+export class BasicAuthAuthorizer {
+
+    constructor(){
+
+    }
+
+    public getUnauthorizedResponse(req: any) {
+        return req.auth
+            ? ('Credentials ' + req.auth.user + ':' + req.auth.password + ' rejected')
```
Comment (arifalkner) [inline src/classes/BasicAuthAuthorizer.ts:10]: Exposing passwords even if intended for incorrect ones can be a security bug. What if this accidentally invoked in a happy path?

File: src/classes/BookDataClass.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  constructor(db: Database) {
+    this.db = db;
+  }
+
+  // Method to select a book by book_id
+  public async getBookById(bookId: string) {
+    try {
+      const existingBook = await this.db.get(
+        `SELECT * FROM books WHERE book_id = ?`,
```
Comment (arifalkner) [inline src/classes/BookDataClass.ts:14]: Should we add a LIMIT?
```suggestion
        `SELECT * FROM books WHERE book_id = ? LIMIT 1`,
```

File: src/classes/BookDataClass.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    author: string,
+    textContent: string
+  ) {
+    try {
+      await this.db.run(
+        `INSERT INTO books (book_id, title, author, text_content) VALUES (?, ?, ?, ?)`,
+        bookId,
+        title,
+        author,
+        textContent
```
Comment (arifalkner) [inline src/classes/BookDataClass.ts:37]: nit: maybe use k-v to make it easier to follow and track down errors

File: src/classes/util/RootPathHandler.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            res.status(403).send()
+        } else {
+            next(); // If not one of the rejected methods, pass control to the next middleware or route handler
+        }
+    }
+
+        // Method to handle non-existent routes
+        public handleNotFound(req: Request, res: Response): void {
+            res.status(403).send();
+        }
```
Comment (arifalkner) [inline src/classes/util/RootPathHandler.ts:20]: Shouldn't this be a 404?

File: src/routes/book.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import BookDatabase  from '../classes/BookDataClass';
+import BookFetcher from '../classes/BookFetcherClass';
+
+const router = express.Router();
+
+//TODO: setup basic auth
+//basicAuth({users: { 'Project': ' Gutenberg' }, challenge: true, unauthorizedResponse: new BasicAuthAuthorizer().getUnauthorizedResponse}),
+
+router.get('/:book_id', async (req: Request, res: Response) => {
+    const { book_id } = req.params;
```
Comment (arifalkner) [inline src/routes/book.ts:12]: nit: maintain camelcase
```suggestion
    const { book_id: bookId } = req.params;
```

File: src/classes/BasicAuthAuthorizer.ts (hunk 1 above)
Comment (allynalford) [reply src/classes/BasicAuthAuthorizer.ts:10]: This code isn't being used, i was going to add basic auth.

File: src/classes/BookDataClass.ts (hunk 2 above)
Comment (allynalford) [reply src/classes/BookDataClass.ts:14]: No need, as it's using the primary key to only select one value.

File: src/classes/util/RootPathHandler.ts (hunk 4 above)
Comment (allynalford) [reply src/classes/util/RootPathHandler.ts:20]: No, HTTP 403 Forbidden error response code means that the server understood the request but refused to process it.

File: src/routes/book.ts (hunk 5 above)
Comment (allynalford) [reply src/routes/book.ts:12]: This is to distinguish between a param and a variable within the scope. I get you thou.

File: src/classes/BasicAuthAuthorizer.ts (hunk 1 above)
Comment (allynalford) [reply src/classes/BasicAuthAuthorizer.ts:10]: Removed

**Pull request closed without merging**